src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -0,0 +1,106 @@
+package assisted_installer_controller
+
+import (
+	"context"
+	"time"
+
+	"github.com/eranco74/assisted-installer/src/inventory_client"
+	"github.com/eranco74/assisted-installer/src/k8s_client"
+	"github.com/eranco74/assisted-installer/src/ops"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	done = "Done"
+)
+
+// assiste
+
+type ControllerConfig struct {
+	ClusterID string `envconfig:"CLUSTER_ID" required:"true" `
+	Host      string `envconfig:"INVENTORY_HOST" required:"true"`
+	Port      int    `envconfig:"INVENTORY_PORT" required:"true"`
+}
+
+type Controller interface {
+	WaitAndUpdateNodesStatus()
+}
+
+type controller struct {
+	ControllerConfig
+	log                    *logrus.Logger
+	ops                    ops.Ops
+	ic                     inventory_client.InventoryClient
+	kc                     k8s_client.K8SClient
+	assistedInstallerNodes []string
+}
+
+func NewController(log *logrus.Logger, cfg ControllerConfig, ops ops.Ops, ic inventory_client.InventoryClient, kc k8s_client.K8SClient) *controller {
+	return &controller{
+		log:              log,
+		ControllerConfig: cfg,
+		ops:              ops,
+		ic:               ic,
+		kc:               kc,
+	}
+}
+
+func (c *controller) WaitAndUpdateNodesStatus() {
+	c.log.Infof("Waiting till all nodes will join and update status to assisted installer")
+
+	c.getInventoryNodes()
+
+	apiContext, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	nodeNameAndUUID := make(map[string]string)
+	wait.Until(func() {
+		nodes, err := c.getNodes()
+		if err != nil {
```

Comment:
Consider logging the nodes names

Reply:
There is 2 logs where name will appear. No need for another one